test/test_queue.rb

```diff
@@ -24,4 +24,35 @@ def test_flaky_jobs
 
     assert_failures(["./spec/legit_failure_spec.rb[1:3]"], queue)
   end
+
+  def test_fail_fast
```

Comment:
Not sure if this test adds something more than the end-to-end test above. It tests the same things, right? If so, we can drop this completely.